text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,222 @@
+
+- Tier 3 targets provide no guarantees of support.
+- Tier 2 targets will always build, but may not pass tests.
+- Tier 1 targets will always build and pass tests.
+
+Adding a new tier 3 target imposes minimal requirements; we focus primarily on
+avoiding disruption to other ongoing Rust development.
+
+Tier 2 and tier 1 targets place work on the Rust community as a whole, to avoid
+breaking the target. Thus, these tiers require commensurate efforts from the
+maintainers of the target, to demonstrate value and to minimize any disruptions
+to ongoing Rust development.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Requirements for each tier:
+
+## Tier 3 target policy
+
+At this tier, the Rust project provides no official support for a target, so we
+place minimal requirements on the introduction of targets.
+
+- No central decision is required to add a new tier 3 target. Reviewers may
+  always use their own best judgment regarding the quality of work, and the
+  suitability of a target for the Rust project.
+- If a reviewer wishes to consult a broader team for additional guidance, they
+  should contact the compiler team.
+- If a proposed target or target-specific patch substantially changes code
+  shared with other targets (not just target-specific code), the reviewer
+  should consult the compiler team.
+- If the proposer of a target wishes to appeal the rejection of a target, they
+  may contact the compiler team.
+- Tier 3 targets must use naming consistent with any existing targets; for
+  instance, a target for a similar CPU or OS should not gratuitously use an
+  inconsistent name for that CPU or OS. Targets should normally use the same
+  names as used elsewhere in the broader ecosystem (such as in other
+  toolchains), unless they have a very good reason to diverge.
+- Tier 3 targets may have unusual requirements to build or use, but must not
+  create legal issues for the Rust project or for developers who work on those
+  targets.
+- Where possible, tier 3 targets may wish to provide documentation for the Rust
+  community for how to build and run tests for the platform, ideally using
+  emulation.
+- Tier 3 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to maintain the target. In particular,
+  do not post comments (automated or manual) on a PR that suggests a block on
+  the PR based on the target. (A PR author may choose to help with a tier 3
+  target but is not required to.)
+- Patches adding or updating tier 3 targets must not break any existing target.
+- If a tier 3 target shows no signs of activity and has not built for some
+  time, and removing it would improve the quality of the Rust codebase, we may
+  post a PR to remove it; any such PR will be CCed to people who have
+  previously worked on the platform, to check potential interest.
+
+## Tier 2 target policy
```

Comment:
One concerning aspect of e.g. tier 2 targets is that they sometimes may use `unimplemented!()` in the standard library and then place a [burden](https://github.com/rust-lang/rust/pull/56410#issuecomment-516544572) on PRs such as https://github.com/rust-lang/rust/pull/56410. It would be good to discuss the use of `unimplemented!()`.

Reply:
Added requirements to this effect.